File: Q5/Q5.cpp
/*Check if a number is palindrome or not*/
#include <iostream>
#include <string>
using namespace std;

bool Palindrome(const string& str)
{
    int left = 0;
    int right = str.length() - 1;
    
    while (left < right) 
    {
        if (str[left] != str[right]) 
        {
            return false; 
        }
        left++;
        right--;
    }
    return true; 
}

int main() 
{
    string str;
    
    cout << "Enter a string: ";
    cin >> str; 
    
    if (Palindrome(str)) 
    {
        cout << "The string is a palindrome." << endl;
    } 
    
    else 
    {
        cout << "The string is not a palindrome." << endl;
    }
    
    return 0;
}

File: Q5/Q5.go
//Check if a number is palindrome or not in go 
package main

import (
	"fmt"
)

func isPalindrome(n int) bool {
	original := n
	reversed := 0

	for n > 0 {
		digit := n % 10
		reversed = reversed*10 + digit
		n /= 10
	}

	return original == reversed
}

func main() {
	var num int
	fmt.Print("Enter a number: ")
	fmt.Scanln(&num)

	if isPalindrome(num) {
		fmt.Println(num, "is a palindrome")
	} else {
		fmt.Println(num, "is not a palindrome")
	}
}
